src/firefly/java/edu/caltech/ipac/firefly/server/WebDAVWorkspaceManager.java

```diff
@@ -407,6 +407,25 @@ public WsResponse createParent(String newRelPath) throws WsException {
         return WsUtil.success(200, "Created", getResourceUrl(newRelPath));
     }
 
+    public WsResponse moveFile(String originalFileRelPath, String newPath, boolean overwrite) throws WsException {
+        WspaceMeta meta = new WspaceMeta(newPath);
+        String parent = meta.getParentPath();
+
+        //WARNING: WEBDAv forces me to create new parent first! UGLY!
+        createParent(parent);
+
+        String newUrl = getResourceUrl(newPath);
+
+        // This doesn't create parent , you must create the parent folder in order to move it
+        MoveMethod move = new MoveMethod(getResourceUrl(originalFileRelPath), newUrl, overwrite);
+        if (!executeMethod(move)) {
+            // handle error
+            LOG.error("Unable to move:" + originalFileRelPath + " -- " + move.getStatusText());
```

Comment:
I am glad to see you take the time to log informative errors.  This will come handy when debugging in operation.  newUrl is not important?

Reply:
added.